src/pages/index.js

```diff
@@ -0,0 +1,329 @@
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "save";
+    });
+}
+
+let selectedCard, selectedCardId;
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Deleting...";
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {})
+    .catch(console.error);
+  const button = evt.target.querySelector(".modal__submit-button");
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      console.log("Delete");
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "Delete";
+    });
+}
+
+function handleDeleteCard(element, cardId) {
+  console.log(cardId);
+  selectedCard = element;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleCloseModalByEsc(e) {
+  if (editModal && e.key === "Escape") {
+    const editModal = document.querySelector(".modal_is-opened");
+    closeModal(editModal);
+  }
+}
+function openModal(modal) {
+  modal.classList.add("modal_is-opened");
+
+  document.addEventListener("keydown", handleCloseModalByEsc);
+  modal.addEventListener("mousedown", handleCloseModalOverlayClick);
+}
+function closeModal(modal) {
+  modal.classList.remove("modal_is-opened");
+  document.removeEventListener("keydown", handleCloseModalByEsc);
+  modal.removeEventListener("mousedown", handleCloseModalOverlayClick);
+}
+
+function handleEditFormSubmit(evt) {
+  // const formData = get;
+
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = editModalNameInput.value;
+      profileDescription.textContent = editModalDescriptionInput.value;
+      closeModal(editModal, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "save";
+    });
+}
+
+function handleCloseModalOverlayClick(e) {
+  if (e.target.classList.contains("modal")) closeModal(e.target);
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+
+  const nameInputValue = cardNameInput.value;
+  console.log(nameInputValue);
+
+  const linkInputValue = cardModalLinkInput.value;
+  console.log(linkInputValue);
+
+  const data = {
+    name: cardNameInput.value,
+    link: cardModalLinkInput.value,
+  };
+
+  api
+    .addCard(data)
+    .then((res) => {
+      const cardEl = getCardElement(res);
+      cardsList.prepend(cardEl);
+    })
+    // .catch((error) => {
+    // console.error("error adding card:", error);
+    // });
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "save";
+    });
+
+  closeModal(cardModal);
```

Comment:
You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**